strawberryfields/engine.py

```diff
@@ -335,11 +337,22 @@ def _run_program(self, prog, **kwargs):
         applied = []
         samples_dict = {}
         batches = self.backend_options.get("batch_size", 0)
+
+        # Reset the _all_samples for RegRefs for multiple runs
+        self._all_samples = []
         for cmd in prog.circuit:
             try:
                 # try to apply it to the backend and, if op is a measurement, store it in values
                 val = cmd.op.apply(cmd.reg, self.backend, **kwargs)
                 if val is not None:
+
+                    # Store the measurement outcomes such that not only the last outcome is retrievable
+                    for ind, reg in enumerate(cmd.reg):
+                        # Retrieve the index and the value
+                        # Values are stored in a nested sequence
+                        tup = (reg.ind, val[0][ind])
+                        self._all_samples.append(tup)
+
                     for i, r in enumerate(cmd.reg):
                         if batches:
                             samples_dict[r.ind] = val[:, :, i]
```

Comment:
This can probably be combined into the same for-loop. Also, this would only work for single-shot, and not at all with batches, right? Since you're only choosing the first index of `val`.

```suggestion
                    for i, r in enumerate(cmd.reg):
                        # Retrieve the index and the value
                        # Values are stored in a nested sequence
                        tup = (reg.ind, val[0][ind])
                        self._all_samples.append(tup)
                        
                        if batches:
                            samples_dict[r.ind] = val[:, :, i]
```

Reply:
Oh, this slipped my mind entirely! :sweat_smile: thank you :heart_eyes: :blush: